### PR TITLE
Quiz work

### PR DESCRIPTION
**Pull request by mingchai**
Hey Brandon,

For this one, my authorization method wasn't working for some reason. When you review this, can you advise why that might be happening? I just used a manual work-around for the relevant sections.

Also, for the DOM-Manipulation, I couldn't quite figure out how to keep the JS file separate and instead included the logic on the page for the time being, though I know this isn't the best practice :P

### CODE REVIEW

**Comment by mingchai** [comment]
Git reset done as fix for authorization seems to have caused some tests to fail.

File: app/models/idea.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Idea < ApplicationRecord
+    belongs_to :user
+    has_many :reviews
```
Comment (QuickStyles) [inline app/models/idea.rb:3]: Line 
```ruby
has_many :reviews
```
should be something like:
```ruby
has_many :reviews, dependent: :destroy
```
To gracefully destroy associated Reviews on destruction of an Idea

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class User < ApplicationRecord
+  has_many :ideas, dependent: :nullify
```
Comment (QuickStyles) [inline app/models/user.rb:2]: There should be an association between User and Reviews as well

File: app/controllers/reviews_controller.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    end
+
+    private
+
+    def review_params
+        params.require(:review).permit(:title)
+    end
+
+    def authorize_user!
+        unless can?(:crud, @review)
```
Comment (QuickStyles) [inline app/controllers/reviews_controller.rb:34]: The problem with authorize_user! method here is that @review is nil. You forgot to query it `@review = Review.find(params.require(:id).permit(:id))` within the method OR better practice define a method in app/models/review which can then be evoked on the before_action filter

> app/controllers/reviews_controller.rb
```ruby
before_action Review.get_review_from_params
```

>app/models/review
```ruby
def get_review_from_params
@review = Review.find(params.require(:id).permit(:id))
end
```

**Review by QuickStyles — COMMENTED** [review]
Ming Chai [Quiz 2]
`SCORE 92%/100%`

### Part 1: CRUD
### Value: 30%
### Score: 30% / 30%

> Implement full CRUD for ideas (not fully shown in the wireframe above)
• New, Create, Show, Index, Delete, Edit & Update
`15/15%`

> Each idea has a title and a description.
`10/10%`

> Make sure that there are 6 tests added: 
• 2 for the New action
• 4 for the Create action
`5/5%`

### Part 2: User Authentication
### Value: 30%
### Score: 30/30%

> Implement basic user authentication:
> • Sign Up
> • Sign In
> • Sign Out
`10/10%`

Associate created ideas with current user. 
`10/10%`

Allow only the creator to edit & delete their ideas.
`8/8%`

Update your tests with user auth support.
**Add tests with context of signed in user**
`2/2%`

### Part 3: Reviews
### Value: 30%
### Score: 22/30%

> Implement the ability for users to put reviews on each idea's show page.
`10/10%`

> User must be signed in in order to create a review and the created ideas get associated with the signed in user.
`10/10%`

> The user who created the review should be able to delete the review. 
`2/10%`

### Part 4: Interactivity
### Value: 10%
### Score: 10/10%

> In the ideas#index, only display the title at first.
`2.5%/2.5%`

> Add a + icon (with a picture or Font Awesome) to the left of the title.
`2.5%/2.5%`

> Clicking the + icon should show the description of the idea and change the icon to a – icon.
`2.5%/2.5%`

> Clicking the – icon should make the description disappear.
`2.5%/2.5%`

### Bonus: New Form in a Modal
### Value: 10%
### Score: 0%/10%

>Make clicking New Idea open a modal containing the form for creating new ideas instead of going to another page.
>You can use Bootstrap's own modal, use a Javascript plugin or implement your own. 
>Submitting the form modal should still properly create a new idea.

**Pull request merged**